lib/rules/no-passed-in-event-handlers.js

```diff
@@ -40,7 +40,53 @@ const EVENT_HANDLER_METHODS = new Set([
   'drop',
 ]);
 
+const DEFAULT_CONFIG = {
+  ignore: [],
+};
+
+function isValidConfigObjectFormat(config) {
+  for (let key in config) {
+    let ignores = config[key];
+    let ignoresIsObject = typeof ignores === 'object';
+
+    if (key === 'ignore' && ignoresIsObject) {
+      for (let ignore in ignores) {
+        let valueIsArray = Array.isArray(ignores[ignore]);
+
+        if (!valueIsArray) {
+          return false;
+        }
+      }
+    }
+  }
+
+  return true;
+}
+
 module.exports = class NoPassedInEventHandlers extends Rule {
+  parseConfig(config) {
+    switch (typeof config) {
+      case 'boolean': {
+        return config ? DEFAULT_CONFIG : false;
+      }
+      case 'object': {
+        return isValidConfigObjectFormat(config) ? config : DEFAULT_CONFIG;
```

Comment:
When we aren't valid config wise, we should throw an error (not fallback to default).

Reply:
I think returned false makes sense here, and is inline with what other rules do.